test/unit/widgets/peoplePicker.test.ts

```diff
@@ -5,21 +5,112 @@ import {
+const fetcher = kb.fetcher
 
 const dom = new JSDOM('<!DOCTYPE html><p>Hello world</p>').window.document
 const element = dom.createElement('div')
 
+describe('FindAddressBook', () => {
+  it('exists', () => {
+    expect(findAddressBook).toBeInstanceOf(Function)
+  })
+  it('runs', () => {
+    expect(findAddressBook('typeIndex')).toMatchObject({})
+  })
+
+  it('should call kb.any and spy load when callback is successful ', () => {
+    const spyOnNowOrWhenFetched = jest.spyOn(fetcher, 'nowOrWhenFetched')
+
+    const spyAny = jest
+      .spyOn(kb, 'any')
+      .mockReturnValueOnce('book')
+      .mockReturnValueOnce('book')
+    const spyLoad = jest.spyOn(fetcher, 'load').mockResolvedValue('book')
+    findAddressBook('typeIndex')
+    const callback: any = spyOnNowOrWhenFetched.mock.calls[0][1]
+
+    callback(true, undefined)
+
+    expect(spyOnNowOrWhenFetched).toBeCalled()
+    expect(spyAny).toBeCalled()
+    expect(spyLoad).toBeCalled()
+  })
+  it('should return an error if it is not okay', () => {
+    const spyOnNowOrWhenFetched = jest.spyOn(fetcher, 'nowOrWhenFetched')
+    const spyAny = jest
+      .spyOn(kb, 'any')
+      .mockReturnValue('book')
+      .mockReturnValueOnce('book')
+    const spyLoad = jest.spyOn(fetcher, 'load').mockResolvedValue('book')
+    findAddressBook('typeIndex')
+    const callback: any = spyOnNowOrWhenFetched.mock.calls[0][1]
+    callback(false, undefined)
+    // expect(spyOnNowOrWhenFetched).toThrowError()
+  })
+  it('should return an error if it does not return a book registration', () => {
+    const spyOnNowOrWhenFetched = jest.spyOn(fetcher, 'nowOrWhenFetched')
+    const spyAny = jest
+      .spyOn(kb, 'any')
+      .mockReturnValueOnce(null)
+      .mockReturnValueOnce('book')
+    const spyLoad = jest.spyOn(fetcher, 'load').mockResolvedValue('book')
+    findAddressBook('typeIndex')
+    const callback: any = spyOnNowOrWhenFetched.mock.calls[0][1]
+    callback(true, undefined)
+    // expect(spyOnNowOrWhenFetched).toThrowError()
+  })
+
+  it('should throw an error when there is no book', () => {
+    const spyOnNowOrWhenFetched = jest.spyOn(fetcher, 'nowOrWhenFetched')
+    const spyAny = jest
+      .spyOn(kb, 'any')
+      .mockReturnValueOnce('bookregs')
+      .mockReturnValueOnce(null)
+    const spyLoad = jest.spyOn(fetcher, 'load').mockResolvedValue('book')
+    findAddressBook('typeIndex')
+    const callback: any = spyOnNowOrWhenFetched.mock.calls[0][1]
+    callback(true, undefined)
+    // expect(spyOnNowOrWhenFetched).toThrowError()
+  })
+  it('should throw an error when there an error with the load', () => {
+    // this was working before I changed the above to mockReturnValueOnce
+    // which I needed to do in order to get that test to take affect...
+    // need to research
+    const spyOnNowOrWhenFetched = jest.spyOn(fetcher, 'nowOrWhenFetched')
+    const spyAny = jest
+      .spyOn(kb, 'any')
+      .mockReturnValue('book')
+      .mockReturnValue('book')
```

Comment:
Just FYI: When I see this chunks of code that looks very similar, it can be tempting to write some helper functionality that abstracts these chunks. Than again, that might make it harder to modify tests later, so it's a difficult line to balance. For now, lets avoid the temptation and live with the redundant chunks of code.